samcli/local/apigw/authorizers/lambda_authorizer.py

```diff
@@ -284,3 +296,156 @@ def _parse_identity_sources(self, identity_sources: List[str]) -> None:
                     self._identity_sources.append(identity_source_validator)
 
                     break
+
+    def is_valid_response(self, response: str, method_arn: str) -> bool:
+        """
+        Validates whether a Lambda authorizer request is authenticated or not.
+
+        Parameters
+        ----------
+        response: str
+            JSON string containing the output from a Lambda authorizer
+
+        Returns
+        -------
+        bool
+            True if the request is properly authenticated
+        """
+        try:
+            json_response = json.loads(response)
+        except ValueError:
+            raise InvalidLambdaAuthorizerResponse(
+                f"Authorizer {self.authorizer_name} return an invalid response payload"
+            )
+
+        if self.payload_version == LambdaAuthorizer.PAYLOAD_V2 and self.use_simple_response:
+            return self._validate_simple_response(json_response)
+
+        return self._validate_iam_response(json_response, method_arn)
+
+    def _validate_iam_response(self, response: dict, method_arn: str) -> bool:
```

Comment:
Example responses can be found here as a reference: https://docs.aws.amazon.com/apigateway/latest/developerguide/http-api-lambda-authorizer.html#http-api-lambda-authorizer.payload-format-response